### DISCUSSION TITLE
Removing/hiding video.add_text after x seconds

### DISCUSSION THREAD

**Discussion by gavin2812** [savonet/liquidsoap#4162] [User Support]
Hi all,

I'm currently showing some metadata on tracks change using .on_metadata and this works as expected.

I would like to hide this metadata after x seconds, but I'm not sure how to accomplish this.

This is the section of code:
```

def print_metadata(m) =
  metadata := m["filename"]
end

main_source.on_metadata(print_metadata)

video.add_text(
	font="/home/common/fonts/Ubuntu-Bold.ttf",
	color=0xffffff, 
	speed=0, 
	x=30, 
	y=585, 
	size=38,
	{!metadata},
	main_source)
```
And the above works as expected and the metadata updates on track change.

What would be the best/easiest/most efficient way to hide this text after x seconds?

I have tried, for example:

```
def print_metadata(m) =
  metadata := m["filename"]
  sleep(10.)
  metadata := ""
end
```

But that didn't work. What would be the best way of doing this?

To summarise, show metadata on screen on track change and then remove it after x seconds.

Thanks in advance.

**Comment by gAlleb** [comment]
Hi, example with 10 seconds, something like this:


```ruby

def print_metadata(m) =
  metadata := m["filename"]
end
def hide_metadata() =
  metadata := ""
end

main_source.on_metadata(print_metadata)
main_source.on_metadata(fun(_) -> thread.run(hide_metadata, delay=10., fast=false))

```

Or one line `main_source.on_metadata(fun(_) -> thread.run({ metadata := ""}, delay=10., fast=false))`

**Reply by gavin2812** [reply]
Awesome! Works great! Thanks.

**Comment by gavin2812** [comment]
Just another quick question... How would I do the same for an image? So to show/hide an image in the same way?

I know how to add an image:

```
main_source = video.add_image(
       width=240,height=240,
       x=10,y=1,
       file="/home/public/logo.png",
       main_source)
```
But I've not been able to temporarily hide it, for example, on metadata change same as the text. With the text it was easier (to just set the text string as empty) but how would I do such a thing with an image?

Thanks in advance.

**Reply by gAlleb** [reply]
Same as with text.

Create a `ref` for an image with blank image (e.g. `blank.png`). `ref` shouldn't be empty:

```ruby
show_image = ref("blank.png")

metadata = ref("")

def print_metadata(m) =
  show_image := "logo.png"
  metadata := m["filename"]
end

def hide_metadata() =
  show_image := "blank.png"
  metadata := ""
end

main_source.on_metadata(print_metadata)

main_source.on_metadata(fun(_) -> thread.run(hide_metadata, delay=15., fast=false))

main_source = video.add_image(
       width=240,height=240,
       x=10,y=1,
       file=show_image,
       main_source)

```

**Reply by gavin2812** [reply]
Ahhhh, I was missing the ref bit (and a few other mistakes!).

Thank again!
